### ISSUE TITLE
TopK figure out why input 5D tensor for dims [0,1,2] doesn't work

### ISSUE THREAD

**Issue by aczajkowskiTT** [tenstorrent/tt-metal#21062]
Given a 5D tensor, with permutations of below shapes, dim[0, 1, 2] produces wrong results.  Figure out why, and generate appropriate LLK ticket or solve topk implementation.
@pytest.mark.parametrize("dim1", [1])
@pytest.mark.parametrize("dim2", [1])
@pytest.mark.parametrize("dim3", [8])
@pytest.mark.parametrize("dim4", [256])
@pytest.mark.parametrize("dim5", [64])
# @pytest.mark.parametrize("dim", [0, 1, 2, 3, 4]) #transpose cannot handle N-D tensor for all dims
@pytest.mark.parametrize("dim", [3, 4])
@pytest.mark.parametrize("k", [17, 32, 64])
@pytest.mark.parametrize("largest", [True])
@pytest.mark.parametrize("dtype", [ttnn.bfloat16, ttnn.bfloat8_b])

**Comment by bbradelTT** [comment]
@aczajkowskiTT Any updates?

**Comment by aczajkowskiTT** [comment]
@bbradelTT I will have bandwidth to look at this in the current week

**Comment by mgajewskiTT** [comment]
Tests were conducted as described in this ticket. An additional test case was created:

```python
@pytest.mark.parametrize("dim1", [1])
@pytest.mark.parametrize("dim2", [1])
@pytest.mark.parametrize("dim3", [8])
@pytest.mark.parametrize("dim4", [256])
@pytest.mark.parametrize("dim5", [64])
@pytest.mark.parametrize("dim", [0, 1, 2, 3, 4])
@pytest.mark.parametrize("k", [17, 32, 64])
@pytest.mark.parametrize("largest", [True])
@pytest.mark.parametrize("dtype", [ttnn.bfloat16, ttnn.bfloat8_b])
def test_topk_5d_tensor(dim1, dim2, dim3, dim4, dim5, dim, k, largest, dtype, device):
    """
    Test topk on 5D tensors.
    Issue: dim[0, 1, 2] produces wrong results.
    This test currently only tests dim[3, 4] which should work.
    """
    torch.manual_seed(2005)
    shape = [dim1, dim2, dim3, dim4, dim5]
    torch_dtype = torch.bfloat16
    input = torch.randn(shape, dtype=torch_dtype) * 0.9

    if k > shape[dim]:
        pytest.skip(f"k={k} is greater than dimension size {shape[dim]} for dim={dim}")

    # PyTorch reference
    pyt_topk_values, pyt_topk_indices = torch.topk(input, k, dim=dim, largest=largest, sorted=True)

    # TTNN implementation
    ttnn_input = ttnn.from_torch(input, dtype, layout=ttnn.Layout.TILE, device=device)

    try:
        ttnn_topk_values, ttnn_topk_indices = ttnn.topk(
            ttnn_input,
            k,
            dim=dim,
            largest=largest,
            sorted=True,
        )
    except Exception as e:
        pytest.fail(f"ttnn.topk raised an exception: {e}")

    # Verify output shapes
    desired_shape = list(shape)
    desired_shape[dim] = k
    assert list(ttnn_topk_values.shape) == desired_shape
    assert list(ttnn_topk_indices.shape) == desired_shape

    # Convert back to torch
    ttnn_torch_values = ttnn.to_torch(ttnn_topk_values)
    ttnn_torch_indices = ttnn.to_torch(ttnn_topk_indices).to(torch.int64)

    # Check values with PCC
    pcc_values = 0.99 if dtype == ttnn.bfloat8_b else 1.0

    # Cosine similarity on gathered values to handle ties
    ttnn_torch_gather_from_indices = torch.gather(input, dim, ttnn_torch_indices)
    cosine = torch.nn.CosineSimilarity(dim=dim)
    ttnn_torch_cosine = torch.mean(cosine(pyt_topk_values, ttnn_torch_gather_from_indices))

    assert ttnn_torch_cosine > 0.99
    assert_with_pcc(pyt_topk_values, ttnn_torch_values, pcc_values)
```

The thing is: everything passes. I’m not seeing any incorrect behavior.
It’s likely the 5D tensor issue was fixed incidentally by a later change.

I’m closing this ticket with these findings. If the issue is still relevant, or if it was meant to cover something else, please reopen it.
